site/docs/v1/tech/apis/webauthn.adoc

```diff
@@ -146,9 +149,47 @@ Only a status code is available on the Import API, no JSON body will be returned
 
+== WebAuthn JavaScript API Binary Format
+The https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API[WebAuthn JavaScript API] `navigator.credentials.create()` and `navigator.credentials.get()` expect to receive fields containing binary data on the `options` object as a JavaScript https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer[``ArrayBuffer``] and will return binary fields as ``ArrayBuffer``s. In order to prevent encoding issues on the FusionAuth API, these fields are passed over the network as base64url-encoded strings.
+
+Select fields on the `options` JSON object that is passed to the https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API[WebAuthn JavaScript API] must be converted from base64url-encoded strings to ``ArrayBuffer``s after receiving `options` from the FusionAuth API. Likewise, certain fields on https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API[WebAuthn JavaScript API] responses must be converted from ``ArrayBuffer``s to base64url-encoded strings before calling FusionAuth's APIs to complete the ceremony.
+
+=== Converting base64url-encoded String to `ArrayBuffer`
+Converting a based64url-encoded strings to ``ArrayBuffer``s is required before the `options` JSON object from <<Start a WebAuthn Passkey Registration>> or <<Start a WebAuthn Passkey Assertion or Authentication>> responses are passed to the https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API[WebAuthn JavaScript API]. The FusionAuth hosted pages will perform this conversion as necessary. If you need to perform this conversion yourself, you can use the following JavaScript function.
+
+[source,javascript]
+.Function to convert a base64url-encoded string to an `ArrayBuffer`
+----
+function base64URLToBuffer(base64URL) {
+    const base64 = base64URL.replace(/-/g, '+').replace(/_/g, '/');
+    const padLen = (4 - (base64.length % 4)) % 4;
+    return Uint8Array.from(atob(base64.padEnd(base64.length + padLen, '=')), c => c.charCodeAt(0));
+}
+----
+
+Fields that require this conversion are documented in the <<Start a WebAuthn Passkey Registration>> and <<Start a WebAuthn Passkey Assertion or Authentication>> response sections.
+
+=== Converting `ArrayBuffer` to base64url-encoded String
+Converting ``ArrayBuffer``s to base64url-encoded strings is required before the responses from the https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API[WebAuthn JavaScript APIs] are sent to FusionAuth's <<Complete a WebAuthn Passkey Registration>>, <<Complete a WebAuthn Passkey Authentication>>, or <<Complete a WebAuthn Passkey Assertion>> APIs. The FusionAuth hosted pages will perform this conversion as necessary. If you need to perform this conversion yourself, you can use the following JavaScript function.
+
+[source,javascript]
+.Function to convert an `ArrayBuffer` to a base64url-encoded string
+----
+function bufferToBase64URL(buffer) {
```

Comment:
These are great. Only feedback would be to put them into a separate repo so that they can be unit tested by a GH action. You can import values from a separate repo like so:

```
[source,yaml,title=docker-compose.yml]
----
include::https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/docker-compose.yml[]
----
```

But that's a nit.

Reply:
Do we have a repo that would fit into?